### PR TITLE
v4.0.0-alpha.36

### PR DESCRIPTION
**Pull request by DanailH**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,66 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 4.0.0-alpha.36
+
+_August 5, 2021_
+
+Big thanks to the 4 contributors who made this release possible. Here are some highlights ✨:
+
+- 🚀 Polish the cell editing api (#2220) @m4theushw
```
Comment (m4theushw) [inline CHANGELOG.md:12]:
```suggestion
- 🚀 Polish the cell editing API (#2220) @m4theushw
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,66 @@
...
+
+### `@material-ui/data-grid@v4.0.0-alpha.36` / `@material-ui/x-grid@v4.0.0-alpha.36`
+
+#### Breaking changes
+
+- [DataGrid] Polish cell editing (#2220) @m4theushw
+
+  - [DataGrid] The `onCellModeChange` prop was replaced with `onCellEditEnter` or `onCellEditExit`.
+  - [DataGrid] The `onCellEditEnter` prop was renamed to `onCellEditStart`.
+  - [DataGrid] The `onCellEditEnd` prop was renamed to `onCellEditStop`.
```
Comment (m4theushw) [inline CHANGELOG.md:25]:
```suggestion
  - Replace `onCellModeChange` prop with `onCellEditEnter` or `onCellEditExit`.
  - Rename `onCellEditEnter` prop to `onCellEditStart`.
  - Rename `onCellEditEnd` prop to `onCellEditStop`.

```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,66 @@
...
+ ```diff
+  <DataGrid
+  - onCellEditEnter={...}
+  - onCellEditExit={...}
+  + onCellEditStart={...}
+  + onCellEditStop={...}
+  />
+  ```
+
+  - [XGrid] The `setEditCellProps` API call is not available anymore. Use the [controlled editing](https://material-ui.com/components/data-grid/editing/#controlled-editing) or `setEditRowsModel`.
```
Comment (m4theushw) [inline CHANGELOG.md:35]:
```suggestion
  - [XGrid] The `setEditCellProps` API call is not available anymore. 
    Use the [controlled editing](https://material-ui.com/components/data-grid/editing/#controlled-editing) or `setEditRowsModel`.

```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,66 @@
...
+
+- [docs] Improve slot API docs (#2219) @oliviertassinari
+
+### Core
+
+- [core] Isolate `DataGrid` and `XGrid` (#2176) @dtassone
+- [core] Move `GridFilterModel` in the models directory (#2243) @flaviendelangle
+- [core] Added new column internal `computedWidth` field (#2235) @flaviendelangle
+- [core] Use `rootProps` instead of `options` in the grid components except for `classes` (#2213) @flaviendelangle
+- [core] `rebaseWhen=auto` not working (#2271) @oliviertassinari
```
Comment (m4theushw) [inline CHANGELOG.md:64]:
```suggestion
- [core] Fix `rebaseWhen=auto` not working (#2271) @oliviertassinari
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,66 @@
...
 
+## 4.0.0-alpha.36
+
+_August 5, 2021_
+
+Big thanks to the 4 contributors who made this release possible. Here are some highlights ✨:
+
+- 🚀 Polish the cell editing api (#2220) @m4theushw
+- ⚡️ Add `details` param to each callback option in `XGrid` (#2236) @DanailH
+- 💅 Work on internal optimisations and code separation (#2176, #2243, #2235, #2213) @flaviendelangle
```
Comment (m4theushw) [inline CHANGELOG.md:14]:
```suggestion
- 💅 Work on internal optimizations and code separation (#2176, #2243, #2235, #2213) @flaviendelangle
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,66 @@
...
+- 🐞 Fix 1 bugs
```
Comment (m4theushw) [inline CHANGELOG.md:15]:
```suggestion
- 🐞 Fix one bug related to filtering
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,69 @@
...
+  - onCellEditExit={...}
+  + onCellEditStart={...}
+  + onCellEditStop={...}
+  />
+  ```
+
+  - [XGrid] The `setEditCellProps` API call is not available anymore. 
+    Use the [controlled editing](https://material-ui.com/components/data-grid/editing/#controlled-editing) or `setEditRowsModel`.
+
+ ```diff
```
Comment (m4theushw) [inline CHANGELOG.md:39]:
```suggestion
  ```diff
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,69 @@
...
+
+#### Breaking changes
+
+- [DataGrid] Polish cell editing (#2220) @m4theushw
+
+  - Replace `onCellModeChange` prop with `onCellEditEnter` or `onCellEditExit`.
+  - Rename `onCellEditEnter` prop to `onCellEditStart`.
+  - Rename `onCellEditEnd` prop to `onCellEditStop`.
+
+ ```diff
```
Comment (m4theushw) [inline CHANGELOG.md:27]: Indentation.

```suggestion
  ```diff
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,69 @@
...
+  <DataGrid
```
Comment (oliviertassinari) [inline CHANGELOG.md:28]: Indentation within in the diff block
```suggestion
   <DataGrid
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,69 @@
...
+
+### Docs
+
+- [docs] Improve slot API docs (#2219) @oliviertassinari
+
+### Core
+
+- [core] Isolate `DataGrid` and `XGrid` (#2176) @dtassone
+- [core] Move `GridFilterModel` in the models directory (#2243) @flaviendelangle
+- [core] Added new column internal `computedWidth` field (#2235) @flaviendelangle
```
Comment (oliviertassinari) [inline CHANGELOG.md:65]:
```suggestion
- [core] Add new column internal `computedWidth` field (#2235) @flaviendelangle
```

**Comment by m4theushw** [comment]
We had a few merges since this PR was open.

image

File: CHANGELOG.md [hunk 11]
```diff
@@ -3,6 +3,69 @@
...
+## 4.0.0-alpha.36
+
+_August 5, 2021_
+
+Big thanks to the 4 contributors who made this release possible. Here are some highlights ✨:
+
+- 🚀 Polish the cell editing API (#2220) @m4theushw
+- ⚡️ Add `details` param to each callback option in `XGrid` (#2236) @DanailH
+- 💅 Work on internal optimizations and code separation (#2176, #2243, #2235, #2213) @flaviendelangle
+- 🐞 Fix one bug related to filtering
```
Comment (flaviendelangle) [inline CHANGELOG.md:15]: No PR to link here ?

File: CHANGELOG.md [hunk 12]
```diff
@@ -3,6 +3,69 @@
...
+
+### `@material-ui/data-grid@v4.0.0-alpha.36` / `@material-ui/x-grid@v4.0.0-alpha.36`
+
+#### Breaking changes
+
+- [DataGrid] Polish cell editing (#2220) @m4theushw
+
+  - Replace `onCellModeChange` prop with `onCellEditEnter` or `onCellEditExit`.
```
Comment (flaviendelangle) [inline CHANGELOG.md:23]: I think we should use the new callback names here

`
```suggestion
  - Replace `onCellModeChange` prop with `onCellEditStart` or `onCellEditStop`.
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -3,6 +3,69 @@
...
+
+### Docs
+
+- [docs] Improve slot API docs (#2219) @oliviertassinari
+
+### Core
+
+- [core] Isolate `DataGrid` and `XGrid` (#2176) @dtassone
+- [core] Move `GridFilterModel` in the models directory (#2243) @flaviendelangle
+- [core] Add new column internal `computedWidth` field (#2235) @flaviendelangle
```
Comment (flaviendelangle) [inline CHANGELOG.md:65]: We should probably mark this one as breaking for the callbacks


- The `width` property of the columns is no longer updated with the actual width of of the column. Use the new `computedWidth` property in the callbacks instead.

```diff
const columns: GridColDef = [
  {
    field: "name",
    width: 100,
    renderCell: ({ value, colDef }) => {
-     console.log(colDef.width!)
+     console.log(colDef.computedWidth)      
      return value
    }
]
```

or

```diff
- onColumnOrderChange = ({ colDef }) => console.log(colDef.width!)
+ onColumnOrderChange = ({ colDef }) => console.log(colDef.computedWidth)
```

Comment (DanailH) [reply CHANGELOG.md:65]: It's internal, no? why should it be breaking change?

File: CHANGELOG.md (hunk 12 above)
Comment (DanailH) [reply CHANGELOG.md:23]: Not sure. I would keep it as it was explained in the PR description

File: CHANGELOG.md (hunk 11 above)
Comment (DanailH) [reply CHANGELOG.md:15]: We usually don't do PR links for bugs.

File: CHANGELOG.md (hunk 13 above)
Comment (flaviendelangle) [reply CHANGELOG.md:65]: It's not only internal, you can see in my two examples how is changes the behavior in the user code
You have an example in the `RenderExpandCellGrid` demo.

Comment (DanailH) [reply CHANGELOG.md:69]: Wait what? So the user provides `width` initially but then gets `computedWidth`?

Comment (DanailH) [reply CHANGELOG.md:85]: Updated

File: CHANGELOG.md (hunk 12 above)
Comment (oliviertassinari) [reply CHANGELOG.md:24]: `onCellEditEnter` does no longer exist

**Comment by DanailH** [comment]
@mui-org/x If no one minds I'll proceed with merging this and finish the release.

**Review by flaviendelangle — APPROVED** [review]

File: CHANGELOG.md (hunk 12 above)
Comment (oliviertassinari) [reply CHANGELOG.md]: I have updated https://github.com/mui-org/material-ui-x/pull/2220 to fix it there too

**Pull request merged**